rustls/src/client/tls13.rs

```diff
@@ -392,6 +421,18 @@ impl State<ClientConnectionData> for ExpectEncryptedExtensions {
         validate_encrypted_extensions(cx.common, &self.hello, exts)?;
         hs::process_alpn_protocol(cx.common, &self.config, exts.get_alpn_protocol())?;
 
+        // Now that we have the server extensions we need to check if our ECH offer was rejected.
+        // If this is the case we abort with the ECH required alert. However, we do this at
+        // this stage rather than in the ExpectServerHello state because we want to be able to
+        // include retry configs that may be present in the server's encrypted extensions. This
+        // also allows us to send the alert in encrypted form.
+        if matches!(self.ech_status, ech::Status::Rejected) {
+            let retry_configs = exts
+                .get_server_ech_extension()
+                .map(|ext| ext.retry_configs);
+            return Err(ech::fatal_alert_required(cx.common, retry_configs));
```

Comment:
Could `ech::fatal_alert_required()` just take `&exts` instead?

Reply:
No, I don't think that will work well because we're currently using `fatal_alert_required` in two places:

1. From `confirm_hrr_acceptance` via `ExpectServerHelloOrHelloRetryRequest` when we discover ECH wasn't accepted during a hello retry request.
2. From `handle` in `State<ClientConnectionData> for ExpectEncryptedExtensions`.

In the second place we have a `exts` to pass through, but in the first place we don't have any server extensions yet and so use `None`.

Reply:
Fair. Maybe consider just inlining the binding.

Reply:
Done.